packages/babel-parser/src/tokenizer/types.js

```diff
@@ -41,7 +41,13 @@ type TokenOptions = {
   binop?: ?number,
 };
 
-export class TokenType {
+// Internally the tokenizer stores token as a number
+export type TokenType = number;
```

Comment:
Would it make sense to do this?
```suggestion
export opaque type TokenType = number;
```